controllers/notificationpolicy_controller.go

```diff
@@ -99,6 +116,13 @@ func (r *GrafanaNotificationPolicyReconciler) Reconcile(ctx context.Context, req
 		}
 	}()
 
+	// check if spec is valid
+	if !notificationPolicy.Spec.Route.IsRouteSelectorMutuallyExclusive() {
+		setInvalidSpecMutuallyExclusive(&notificationPolicy.Status.Conditions, notificationPolicy.Generation)
+		return ctrl.Result{}, nil
```

Comment:
We should return an error for the log in case users miss the status and rely on the log output for monitoring.